yt/data_objects/profiles.py

```diff
@@ -1313,7 +1320,11 @@ def create_profile(
                 data_source.ds.field_info[f].sampling_type == "local"
                 for f in bin_fields + fields
             ]
-            is_local_or_pfield = [pf or lf for (pf, lf) in zip(is_pfield, is_local)]
+            if wf is not None:
+                is_local.append(wf.sampling_type == "local")
```

Comment:
highlighting this bit as a bugfix

Reply:
Thank you.  I have an inkling somewhere in my head that there's an extant issue this could be related to but searching does not prove me correct.